File: Delivery System 2/courier.cpp
#include "courier.h"

// Конструктор
Courier::Courier(int id, const std::string& name, const std::string& phoneNumber, const std::string& vehicleType, bool isAvailable)
	: m_id(id), m_name(name), m_phoneNumber(phoneNumber), m_vehicleType(vehicleType), m_isAvailable(isAvailable) {
}

// Назначение заказа курьеру
bool Courier::assignOrder(std::shared_ptr<Order> order) {
	if (!m_isAvailable) return false;
	m_currentOrders.push_back(order);
	return true;
}

File: Delivery System 2 Java/src/Courier.java
package src;

import java.util.ArrayList;
import java.util.List;

public class Courier {
    private int id;
    private String name;
    private String phoneNumber;
    private String vehicleType;
    private boolean isAvailable;
    private List<Order> currentOrders;

    // Конструктор
    public Courier(int id, String name, String phoneNumber, String vehicleType, boolean isAvailable) {
        this.id = id;
        this.name = name;
        this.phoneNumber = phoneNumber;
        this.vehicleType = vehicleType;
        this.isAvailable = isAvailable;
        this.currentOrders = new ArrayList<>();
    }

    // Методы
    public boolean assignOrder(Order order) {
        if (!isAvailable) return false;
        currentOrders.add(order);
        return true;
    }

    // Геттеры
    public int getId() { return id; }
    public String getName() { return name; }
    public String getPhoneNumber() { return phoneNumber; }
    public String getVehicleType() { return vehicleType; }
    public boolean getIsAvailable() { return isAvailable; }
}